packages/analytics-browser/src/plugins/form-interaction-tracking.ts

```diff
@@ -122,3 +124,15 @@ export const formInteractionTracking = (): EnrichmentPlugin => {
     teardown,
   };
 };
+
+export const stringOrUndefined = (name: string) => {
```

Comment:
wondering if we can set this up to do proper type narrowing?

```ts
export const stringOrUndefined = <T>(
  name: T,
): T extends string ? string : undefined => {
  if (typeof name !== 'string') {
    return undefined as T extends string ? string : undefined;
  }

  return name as T extends string ? string : undefined;
};
```

Reply:
i.e. 
```ts

const str = stringOrUndefined('hello');
// ?^ string
const str2 = stringOrUndefined(undefined);
// ?^ undefined
```

Reply:
ohh i like this!

Reply:
fixed!